analytics/request.py

```diff
@@ -2,29 +2,40 @@
 from dateutil.tz import tzutc
 import logging
 import json
-from analytics.version import VERSION
-
+from gzip import GzipFile
 from requests.auth import HTTPBasicAuth
 from requests import sessions
+try:
+    from StringIO import StringIO
+except ImportError:
+    from io import BytesIO as StringIO
 
+from analytics.version import VERSION
 from analytics.utils import remove_trailing_slash
 
 _session = sessions.Session()
 
 
-def post(write_key, host=None, **kwargs):
+def post(write_key, host=None, gzip=False, **kwargs):
     """Post the `kwargs` to the API"""
     log = logging.getLogger('segment')
     body = kwargs
     body["sentAt"] = datetime.utcnow().replace(tzinfo=tzutc()).isoformat()
     url = remove_trailing_slash(host or 'https://api.segment.io') + '/v1/batch'
     auth = HTTPBasicAuth(write_key, '')
     data = json.dumps(body, cls=DatetimeSerializer)
+    log.debug('making request: %s', data)
```

Comment:
was this left from debugging?

Reply:
this was also just moved up from a few lines below, to work with both compressed and uncompressed data.

Reply:
gotcha!